ArduCopter/AP_Arming.cpp

```diff
@@ -631,7 +631,8 @@ bool AP_Arming_Copter::arm_checks(AP_Arming::Method method)
 
 #ifndef ALLOW_ARM_NO_COMPASS
     // if external source of heading is available, we can skip compass health check
-    if (!ahrs.is_ext_nav_used_for_yaw()) {
+    // skip this check if EKF's yaw source is not compass
+    if (!ahrs.is_ext_nav_used_for_yaw() && ahrs.EKF3.isYaw_Source(AP_NavEKF_Source::SourceYaw::COMPASS)) {
```

Comment:
needs to cope with EKF3 not in the build (use HAL_NAVEKF3_AVAILABLE) and also needs to cope with SourceYaw::GPS_COMPASS_FALLBACK and also needs to check we are actually using EKF3

Reply:
better to add a AP_AHRS::need_compass() call

Reply:
Got it ,sir.I will start working on it immediately.